File: week-04/day-4/08.py
# 8. Given a string, compute recursively a new string where all the 'x' chars have been removed.

text = "abcdex abcdex abcdex abcdex"

def coder(text):
    if len(text) == 0:
        return text
    elif text[0] == 'x' or text[0] == 'X':
        return '' + coder(text[1:])
    else:
        return text[0] + coder(text[1:])

print (coder(text))

File: week-07/day-1/recursion/08.js
// 8. Given a string, compute recursively a new string
// where all the 'x' chars have been removed.

var text = "abcdex abcdex abcdex abcdex"

function removerX(inputText) {
  if (inputText.length == 0) {
    return inputText;
  } else if (inputText[0] == 'x' || inputText[0] == 'X') {
    return '' + removerX(inputText.slice(1, inputText.length));
  } else {
    return inputText[0] + removerX(inputText.slice(1, inputText.length));
  }
}

console.log(removerX(text))
